lib/factory_trace/preprocessors/extract_used.rb

```diff
@@ -9,8 +9,11 @@ class ExtractUsed
       def self.call(trace)
         collection = FactoryTrace::Structures::Collection.new
 
-        trace.each do |factory_name, trait_names|
-          collection.add(FactoryTrace::Structures::Factory.new(factory_name, nil, trait_names.to_a))
+        trace.each do |factory_name, factory_data|
```

Comment:
Revert this part please too.

Reply:
With reverting it issue came back :) I see you don't see the problem. Ok. Fix it yourself.

Reply:
Let me try and I share the PR with you and we can discuss. 

Sorry, if I disappointed you, don't take it personally, your reaction surprises me.

Reply:
I'm not disappointed, I just don't want you spend your time on my bad code.

Reply:
Okay, let me try and I need your help to see what is my suggestion. I started just now and will share a PR soon.

Reply:
Okay, let me try and I need your help to see what is my suggestion. I started just now and will share a PR soon.